### PR TITLE
Run `@db_access` tests on CircleCI, support container parallelization > 2

### PR DESCRIPTION
**Pull request by bcjordan**
Resolves the known issue / feature need: [@db_access tests should also be run on CircleCI](http://wiki.code.org/display/PROD/Known+CircleCI+Issues#KnownCircleCIIssues-@db_accesstests)
1. Because this increased container #1 run time to from ~50-60 minutes to ~1h20, this PR adds support to avoid increasing total test time, resulting in a [successful run ~55 minutes with 4 containers](https://circleci.com/gh/code-dot-org/code-dot-org/21014) (now bound by the unit test container)
2. This squashes a few issues related to subdomain handling for UI tests. Because CircleCI runs with the "test" rails env, certain assumptions about the test host had to be worked around.
   - [Add configurable setting to not assume https sessions store.](https://github.com/code-dot-org/code-dot-org/compare/staging...test-db-circleci-3#diff-405ab698f56d3645c57368dcda136673R5)
   - Pegasus resolution middleware (serve pegasus when subdomain is localhost.code.org:3000) had to be [changed to ignore the `:3000` portion of the request host during check](https://github.com/code-dot-org/code-dot-org/compare/staging...test-db-circleci-3#diff-d012b82d1c36e6c8c6078f295031d0f3R29). Technique recommended by @wjordan.
3. A few tests with remaining issues after these changes were marked `@no_circle` temporarily, to be investigated later ([tracking & details](http://wiki.code.org/display/PROD/Known+CircleCI+Issues#KnownCircleCIIssues-Certaintestsaremarked@no_circle) in Known CircleCI issues).

Roll-out plan:
- [ ] Monitor staging, test runs for Eyes changes ([currently not run on feature branches](http://wiki.code.org/display/PROD/Known+CircleCI+Issues#KnownCircleCIIssues-EyesTestsonFeatureBranches)). If many Eyes failures on staging due to not properly falling back to parent branch tests's baselines, merge eyes baselines test -> staging with `rake eyes:merge[test,staging]` so that Eyes tests which have previously never run on staging will pass on staging.
- [ ] Monitor build times and flakiness, balancing container # with run time. If flakiness or run time still regressed, revert and optimize further before merging.

### CODE REVIEW

**Comment by bcjordan** [comment]
Running perf test with 3 different container parallelizations:
1. [2 parallel](https://circleci.com/gh/code-dot-org/code-dot-org/21015)
2. [3 parallel](https://circleci.com/gh/code-dot-org/code-dot-org/21016)
3. [4 parallel](https://circleci.com/gh/code-dot-org/code-dot-org/21017)

File: dashboard/config/initializers/session_store.rb [hunk 1]
```diff
@@ -2,5 +2,5 @@
 session_cookie_key += "_#{Rails.env}" unless Rails.env.production?
 Dashboard::Application.config.session_store :cookie_store,
   key: session_cookie_key,
-  secure: !Rails.env.development? || CDO.https_development,
+  secure: (!CDO.no_https_store) && (!Rails.env.development? || CDO.https_development),
```
Comment (ashercodeorg) [inline dashboard/config/initializers/session_store.rb]: nit: Extra parens not necessary around `!CDO.no_https_store`.

**Comment by ashercodeorg** [comment]
Don't think I'm the right person to review this, despite being awake. :(

**Comment by wjordan** [comment]
LGTM with a quick scan through the changes diff. Since these are all incremental changes/tweaks to test-runner logic I haven't really spent too much time in myself, I don't know the surrounding context well enough to offer a more detailed review but I can at least confirm that none of the changes look too scary or would break anything unexpected, and since you're monitoring and testing these changes as you're rolling them out, that should be good enough :shipit:

**Pull request merged**